### PR TITLE
Handle subfields in the $project stage of the aggregate pipeline.

### LINKED ISSUE #397: Aggregation framework $project does not work on fields within nested documents
When using {'$projection' : {'doc_field.subfield': 1}} in an aggregation pipeline, this doesn't work:
  File ".../lib/python2.7/site-packages/mongomock/collection.py", line 1627, in aggregate
    out_collection = _extend_collection(out_collection, field, value)
  File ".../lib/python2.7/site-packages/mongomock/collection.py", line 1506, in _extend_collection
    doc[field] = _parse_expression(expression.copy(), doc)
AttributeError: 'int' object has no attribute 'copy'

### PR DESCRIPTION
**Pull request by pcorpet**
Fix #397.

### CODE REVIEW

**Comment by boxmansf** [comment]
@pcorpet Basically, looks good to me, but the code is rather complex and I am not very familiar with it. Do you think we could benefit from another review by someone who made changes in these areas ?

File: mongomock/aggregate.py [hunk 1]
```diff
@@ -585,6 +586,63 @@ def _handle_unwind_stage(in_collection, unused_database, options):
     return unwound_collection
 
 
+# TODO(pascal): Combine with the equivalent function in collection but check
+# what are the allowed overriding.
+def _combine_projection_spec(filter_list, original_filter, prefix=''):
+    """Re-format a projection fields spec into a nested dictionary.
+
+    e.g: {'a': 1, 'b.c': 1, 'b.d': 1} => {'a': 1, 'b': {'c': 1, 'd': 1}}
```
Comment (zozoens31) [inline mongomock/aggregate.py]: Maybe add that input `filter_list` is the list of keys to keep, not the $project object itself

**Review by zozoens31 — APPROVED** [review]
LGTM

File: mongomock/aggregate.py (hunk 1 above)
Comment (pcorpet) [reply mongomock/aggregate.py]: Done.

**Pull request merged**